api/client/commands.go

```diff
@@ -2660,6 +2666,10 @@ func (s *containerStats) Collect(cli *DockerCli) {
 		for {
 			var v *types.Stats
 			if err := dec.Decode(&v); err != nil {
+				if err == io.EOF && !streamStats {
```

Comment:
I'm confused about the use of `&& !streamStats` here. If the server breaks after 1 stream anyway, isn't the stop condition properly covered by the `err = io.EOF` part?